### PR TITLE
akka http javadsl - add debugging directive test

### PR DESCRIPTION
**Pull request by fagossa**
Adds a **debugging directive example** to the Java DSL, following the instructions given in [issue 20466](https://github.com/akka/akka/issues/20466)

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.stream.Collectors;
+import java.util.Optional;
+
+public class DebuggingDirectivesExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testLogRequest() {
+    //#logRequest
+    // logs request with "get-user"
+    final Route routeBasicLogRequest = get(() ->
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java]: Please use an indentation of 2 spaces for every level, the docs is cramped column wise, so to keep it easy to read we have diverged from "regular" java indentation practices which is takes lots of column-space.

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  }
+
+  @Test
+  public void testLogRequestResult() {
+    //#logRequestResult
+    // using logRequestResult
+
+    // handle request to optionally generate a log entry
+    BiFunction<HttpRequest, HttpResponse, Optional<LogEntry>> requestMethodAsInfo = 
+      (request, response)->
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java]: space between parameter list and arrow

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        (response.status().isSuccess())  ? 
+            Optional.of(
+              LogEntry
+                .create(request.method().toString() + ":" + response.status().intValue(),
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java]: Funky line-break here, shouldn't it be `LogEntry.create(LF` just like `Optional.of(LF` above?

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            )
+          : Optional.empty(); // not a successful response
+
+    // handle rejections to optionally generate a log entry
+    BiFunction<HttpRequest, List<Rejection>, Optional<LogEntry>> rejectionsAsInfo = 
+      (request, rejections)->
+        (!rejections.isEmpty())  ? 
+          Optional.of(
+            LogEntry.create(rejections
+                          .stream()
```
Comment (johanandren) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java]: 2-space these as well

**Comment by johanandren** [comment]
LGTM except for nitpicks about formatting that makes it look better in the docs

**Comment by drewhk** [comment]
LGTM, thank you for the contribution?

**Comment by fagossa** [comment]
@johanandren @drewhk I've just commited an update improving line breaks according to previous comments

File: akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      (request, rejections) ->
+        (!rejections.isEmpty())  ? 
+          Optional.of(
+            LogEntry.create(
+                rejections
+                .stream()
+                .map(Rejection::toString)
+                .collect(Collectors.joining(", ")), 
+              InfoLevel()))
+          : Optional.empty(); // no rejections
```
Comment (ktoso) [inline akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java]: wouldn't these be easier to read using Optional's orElseGet?  https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseGet-java.util.function.Supplier-

**Comment by ktoso** [comment]
LGTM, thanks a lot @fagossa !

If you have a second to spare rewriting the two examples that use the `?:` operator to use orElseGet would be a very nice follow up PR, thanks in advance! https://github.com/akka/akka/pull/20651#discussion_r65042377

**Pull request merged**